SSA/Projects/InstCombine/ForLean.lean

```diff
@@ -569,6 +569,25 @@ theorem xor_allOnes_sshiftRight_xor_allOnes {a b : BitVec w} :
     rw [ih]
     rw [BitVec.sshiftRight_add]
 
+@[simp]
+theorem shiftLeft_shiftRight {x : BitVec w} {n : Nat}:
+  x >>> n <<< n = x &&& BitVec.allOnes w <<< n := by
+  induction n generalizing x
+  case zero =>
+    ext; simp
+  case succ n ih =>
+    rw [BitVec.shiftLeft_add, Nat.add_comm, BitVec.shiftRight_add, ih,
+       Nat.add_comm, BitVec.shiftLeft_add, BitVec.shiftLeft_and_distrib]
+    ext i
+    simp
+    rw [Nat.add_comm]
```

Comment:
```suggestion
    simp only [getLsbD_and, getLsbD_shiftLeft, Fin.is_lt, decide_True, Nat.lt_one_iff,
      Bool.true_and, getLsbD_ushiftRight, getLsbD_allOnes]
    rw [Nat.add_comm]
```

You get this just by writing `simp?` and then clicking on the blue info text on the right.